draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -130,11 +140,138 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
+
+~~~ json
+{
+ "iss": "wimse://example.com/trusted-central-authority",
+ "exp": 1717612470,
+ "sub": "wimse://example.com/specific-workload",
+ "jti": "x-_1CTL2cca3CSE4cwb__",
+ "cnf": {
+  "jwk": {
+   "kty": "OKP",
+   "crv": "Ed25519",
+   "x": "_amRC3YrYbHhH1RtYrL8cSmTDMhYtOUTG78cGTR5ezk"
+  }
+ }
+}
+~~~
+{: title="Example WIT Claims"}
+
+The claims indicate that the example WIT:
+
+* was issued by an Identity Server known as `wimse://example.com/trusted-central-authority`.
+* is valid until May 15, 2024 3:28:45 PM GMT-06:00 (represented as NumericDate {{Section 2 of RFC7519}} value `1717612470`).
+* identifies the workload to which the token was issued as `wimse://example.com/specific-workload`.
+* has a unique identifier of `x-_1CTL2cca3CSE4cwb__`.
+* binds the public key represented by the `jwk` confirmation method to the workload `wimse://example.com/specific-workload`.
+
+For elucidative purposes only, the workload's key, including the private part, is shown below in JWK {{RFC7517}} format:
+
+~~~ jwk
+{
+ "kty":"OKP",
+ "crv":"Ed25519",
+ "x":"_amRC3YrYbHhH1RtYrL8cSmTDMhYtOUTG78cGTR5ezk",
+ "d":"G4lGAYFtFq5rwyjlgSIRznIoCF7MtKDHByyUUZCqLiA"
+}
+~~~
+{: title="Example Workload's Key"}
+
+The afore-exampled WIT is signed with the private key of the Identity Server.
+The public key(s) of the Identity Server need to be known to all workloads in order to verify the signature of the WIT.
+The Identity Server's public key from this example is shown below in JWK {{RFC7517}} format:
+
+~~~ jwk
+{
+ "kty":"EC",
+ "kid":"June 5",
+ "x":"kXqnA2Op7hgd4zRMbw0iFcc_hDxUxhojxOFVGjE2gks",
+ "y":"n__VndPMR021-59UAs0b9qDTFT-EZtT6xSNs_xFskLo",
+ "crv":"P-256"
+}
+~~~
+{: title="Example Identity Server Key"}
+
+### The WIT HTTP Header {#wit-http-header}
 
-## Option 1: DPoP-Inspired Authentication
+A WIT is conveyed in an HTTP header field named `Workload-Identity-Token`.
 
-## Option 2: Authentication Based on HTTP Message Signatures
+For those who celebrate, ABNF {{RFC5234}} for the value of `Workload-Identity-Token` header field is provided in {{wit-header-abnf}}:
+
+~~~ abnf
+ALPHA = %x41-5A / %x61-7A ; A-Z / a-z
+DIGIT = %x30-39 ; 0-9
+base64url = 1*(ALPHA / DIGIT / "-" / "_")
+WIT =  base64url "." base64url "." base64url
+~~~~
+{: #wit-header-abnf title="Workload-Identity-Token Header Field ABNF"}
+
+The following shows the WIT from the {{example-wit}} in an example of a `Workload-Identity-Token` header field:
+
+~~~ http-message
+Workload-Identity-Token: eyJ0eXAiOiJ3aW1zZS1pZCtqd3QiLCJhbGciOiJFUzI1
+ NiIsImtpZCI6Ikp1bmUgNSJ9.eyJpc3MiOiJ3aW1zZTovL2V4YW1wbGUuY29tL3RydXN
+ 0ZWQtY2VudHJhbC1hdXRob3JpdHkiLCJleHAiOjE3MTc2MTI0NzAsInN1YiI6IndpbXN
+ lOi8vZXhhbXBsZS5jb20vc3BlY2lmaWMtd29ya2xvYWQiLCJqdGkiOiJ4LV8xQ1RMMmN
+ jYTNDU0U0Y3diX18iLCJjbmYiOnsiandrIjp7Imt0eSI6Ik9LUCIsImNydiI6IkVkMjU
+ 1MTkiLCJ4IjoiX2FtUkMzWXJZYkhoSDFSdFlyTDhjU21URE1oWXRPVVRHNzhjR1RSNWV
+ 6ayJ9fX0.rOSUMR8I5WhM5C704l3iVdY0zFqxhugJ8Jo2xo39G7FqUTbwTzAGdpz2lHp
+6eL1M486XmRgl3uyjj6R_iuzNOA
+~~~
```

Comment:
If you want it to be a modern+shiny structured field value (RFC 8941), I think you need colons around the base64 text.

Reply:
OK, this is not SFV. Sad.